weather/module.py

```diff
@@ -98,9 +230,29 @@ async def weather(self, ctx, name: Optional[str] = None):
             await ctx.reply(_(ctx, "You have to specify a place or set a preference."))
             return
 
-        await ctx.reply(name)
+        # try to get user language preference
+        config = pie.database.config.Config.get()
+        lang_preference = None
+        if ctx.guild is not None:
+            # try to get user preference
+            lang_preference = MemberLanguage.get(
+                guild_id=ctx.guild.id, member_id=ctx.author.id
+            )
+            if lang_preference is not None:
+                lang_preference = getattr(lang_preference, "language", None)
+            else:
+                # try to get guild language preference
+                lang_preference = GuildLanguage.get(guild_id=ctx.guild.id)
+                if lang_preference is not None:
+                    lang_preference = getattr(lang_preference, "language", None)
+
+        if lang_preference is None:
+            # set bot language as preference
+            lang_preference = config.language
```

Comment:
This is already done in the `pie.i18n` module. At the top of the file, you can do
```py
translator = i18n.Translator("modules/base")
_ = translator.translate
```
instead of the usual stuff, which will allow you to do
```py
lang_preference: str = translator.get_language_preference(ctx)
```
decreasing the line count from 18 to 1.

Reply:
Fixed, did not know that that method existed.